File: e2e/selenide-junit5-selenium-grid/src/test/java/interaction/step/AccessibilityStep.java
package interaction.step;

import com.codeborne.selenide.WebDriverRunner;
import com.deque.html.axecore.results.Results;
import com.deque.html.axecore.selenium.AxeBuilder;
import io.qameta.allure.Step;

public class AccessibilityStep {

    @Step("Analyze page accessibility with axe")
    public Results analyze() {
        return new AxeBuilder().analyze(WebDriverRunner.getWebDriver());
    }
}

File: demo-android/app/src/androidTest/java/com/example/demo/e2e/interaction/step/AccessibilityStep.kt
package com.example.demo.e2e.interaction.step

import androidx.compose.ui.test.junit4.AndroidComposeTestRule
import androidx.compose.ui.test.onRoot
import androidx.compose.ui.test.tryPerformAccessibilityChecks
import com.example.demo.MainActivity
import io.qameta.allure.kotlin.Allure

class AccessibilityStep(
    private val rule: AndroidComposeTestRule<*, MainActivity>,
) {
    fun analyze() {
        Allure.step("Analyze page accessibility") {
            rule.onRoot().tryPerformAccessibilityChecks()
        }
    }
}
